pkg/networkserver/grpc_asns.go

```diff
@@ -34,7 +34,10 @@ type applicationUpStream struct {
 }
 
 func (s applicationUpStream) Close() error {
-	close(s.closeCh)
+	// First read succeeds when either closeCh is closed by LinkApplication or LinkApplication initializes closing sequence.
+	// Second read succeeds when closeCh is closed by LinkApplication.
+	<-s.closeCh
+	<-s.closeCh
```

Comment:
So `Close()` doesn't close but waits for a close

Reply:
It triggers the "closing sequence" in `LinkApplication`